From 68ac6031d913400b65b2d81b8945c836deaefff1 Mon Sep 17 00:00:00 2001
From: Christian Ambach <ambi@samba.org>
Date: Sun, 3 Apr 2016 05:06:05 +0200
Subject: [PATCH 1052/1096] s3:smbd/service disable case-sensitivity for SMB2/3
 connections

in SMB2, there is no flag to let us know if the client wants to have case-sensitive behavior,
so in Auto mode, disable case-sensitivity

Bug: https://bugzilla.samba.org/show_bug.cgi?id=11438
Signed-off-by: Christian Ambach <ambi@samba.org>
Reviewed-by: Jeremy Allison <jra@samba.org>
(cherry picked from commit 9892c482832b90df28488a7fd4cd1227a0b6ffa5)
---
 source3/smbd/service.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/source3/smbd/service.c b/source3/smbd/service.c
index ba8946d..44c87ed 100644
--- a/source3/smbd/service.c
+++ b/source3/smbd/service.c
@@ -212,7 +212,9 @@ bool set_current_service(connection_struct *conn, uint16_t flags, bool do_chdir)
 			{
 				/* We need this uglyness due to DOS/Win9x clients that lie about case insensitivity. */
 				enum remote_arch_types ra_type = get_remote_arch();
-				if ((ra_type != RA_SAMBA) && (ra_type != RA_CIFSFS)) {
+				if (conn->sconn->using_smb2) {
+					conn->case_sensitive = false;
+				} else if ((ra_type != RA_SAMBA) && (ra_type != RA_CIFSFS)) {
 					/* Client can't support per-packet case sensitive pathnames. */
 					conn->case_sensitive = False;
 				} else {
-- 
2.9.0

